### ISSUE TITLE
`normalize_timestamp': Invalid time value 1453393549

### ISSUE THREAD

**Issue by zinzinzan** [google/signet#70]
hello guys i just install the google api credentials under ruby  and when i try to run my code or try run again ruby quickstart.rb i got the following error:

/home/user/.rvm/gems/ruby-2.1.6/gems/signet-0.7.2/lib/signet/oauth_2/client.rb:1183:in `normalize_timestamp': Invalid time value 1453393549 (RuntimeError)

looks like that value is no a valid date but i dont know that to do that value come from clien_secrent.json generated in the google console

**Comment by zinzinzan** [comment]
hi guys any advice comment? please

**Comment by MikhailSamolinov** [comment]
@zinzinzan I think I just ran into this issue. Could please run `Time.now.to_i.class` in irb to confirm that it returns a Bignum on your system? Here is how it should look if we are talking about the same problem:
```
mikhail$ irb
irb(main):001:0> Time.now.to_i.class
 => Bignum 
```
I've opened a PR (#71) to address the issue. As a quick workaround, you can add [this patch](https://gist.github.com/MikhailSamolinov/635af4fc00e1b46bff04) to your project. In essence, it will replace original implementation of the `normalize_timestamp` with a new version that supports Bignums.

**Comment by zinzinzan** [comment]
thats right bignum is the output
2.1.6 :001 > Time.now.to_i.class
 => Bignum 
2.1.6 :002 > 
where i add that code?? or what i do ?? ruby patch.rb? inside de root folder?

**Comment by zinzinzan** [comment]
i edit the ruby class located in /home/<user>/.rvm/gems/ruby-2.1.6/gems/signet-0.7.2/lib/signet/oauth_2/client.rb

 when Fixnum
          Time.at(time)
        else

to

when Fixnum, Bignum
          #Adding Bignum ^ here as timestamps like 1453983084 are bignums on 32-bit systems
          Time.at(time)
        else

and still give the same error

**Comment by MikhailSamolinov** [comment]
> i edit the ruby class located in /home//.rvm/gems/ruby-2.1.6/gems/signet-0.7.2/lib/signet/oauth_2/client.rb

That certainly works, though I should probably warn you that it will work only until you deploy your code somewhere else as you've patched your local copy of the lib. A better option is to place this patch side by side with the rest of your code and include it before you use the library:
`require 'my_awesome_project/patches.rb'`

**Comment by alexander-js** [comment]
Hi, I rewriting parts of "issued_at" on credential cache json file.

```
credential = "#{path}credential.json"
##
# Google gem bug support.
# signet/client.rb:1172 Bignum is unsupport on normalize_timestamp.
# 'issued_at' rewrite to Time Object String from Bignum.
if File.exist?(credential)
  json = JSON.load(File.read(credential))
  issued = json["issued_at"]
  json["issued_at"] = "#{Time.at(issued)}" if issued.is_a?(Bignum)
  File.open(credential, "w"){|f| f.write JSON.dump(json) }
end
```

**Comment by MikhailSamolinov** [comment]
@alexander-js Yeah, tried that as well. Turned out that in some cases there is no 'issued_at' field in the credentials file and as a result, you can't reliably workaround the issue by patching it.
 
So far I haven't found anything better than [fixing normalize_timestamp at runtime](https://gist.github.com/MikhailSamolinov/635af4fc00e1b46bff04). Hopefully the PR #71 will be merged soon and there will be no need to hack the lib.
  
Here is how my credentials file looks:
```
cat ~/.credentials/com.samolinov.pixtream
---
default: '{
   "client_id":<redacted>,
   "access_token":<redacted>,
   "refresh_token":<redacted>,
   "scope":[
      "https://www.googleapis.com/auth/calendar.readonly"
   ],
   "expiration_time_millis":1453986354000
}
```

**Comment by sqrrrl** [comment]
Fixed by #71
